test/external-entity-references-melinda.spec.js

```diff
@@ -0,0 +1,166 @@
+/**
+ *
+ * @licstart  The following is the entire license notice for the JavaScript code in this file.
+ *
+ * MARC record validators used in Melinda
+ *
+ * Copyright (C) 2014-2018 University Of Helsinki (The National Library Of Finland)
+ *
+ * This file is part of marc-record-validators-melinda
+ *
+ * marc-record-validators-melinda program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * marc-record-validators-melinda is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @licend  The above is the entire license notice
+ * for the JavaScript code in this file.
+ *
+ */
+
+/* eslint-disable no-undef, max-nested-callbacks, no-unused-expressions */
+
+'use strict';
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import MarcRecord from 'marc-record-js';
+import fetchMock from 'fetch-mock';
+import * as testContext from '../src/external-entity-references-melinda'; /* eslint-disable-line import/named */
+import {fixture5000, fixture9550} from './fixtures/external-entity-references';
+
+const {expect} = chai;
+chai.use(chaiAsPromised);
+
+const endpoint = 'http://melinda.kansalliskirjasto.fi:210/fin01';
+const queryParam = '?operation=searchRetrieve&maximumRecords=2&version=1&query=rec.id=';
+
+const prefixPattern = /^\(FOOBAR\)/;
+const fields = {
+	773: ['w'],
+	833: ['w', 'p']
+};
+
+describe('external-entity-references-melinda', () => {
```

Comment:
Add a test for a case where values to resolve are found but are not resolvable (The SRU API returns zero or more than one records).

Reply:
added